### PR TITLE
Set charset for source and execution on Windows

### PR DESCRIPTION
**Pull request by zchrissirhcz**
For the discussion after https://github.com/doxygen/doxygen/pull/10084 is merged.

### CODE REVIEW

File: cmake/WindowsEncoding.cmake [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  execute_process(
+    COMMAND ${PYTHON_EXECUTABLE} "${CMAKE_SOURCE_DIR}/cmake/QueryCodePage.py"
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+    RESULT_VARIABLE ReturnCode
+    OUTPUT_VARIABLE CodePage
+  )
+  message(STATUS "CodePage is ${CodePage}")
+  if("${CodePage}" STREQUAL "936")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /source-charset:utf-8 /execution-charset:gbk")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /source-charset:utf-8 /execution-charset:gbk")
```
Comment (albert-github) [inline cmake/WindowsEncoding.cmake:12]: Does the test case as provided with #10084 still work for the indicated languages when the execution-charset is set to gbk?

Comment (zchrissirhcz) [reply cmake/WindowsEncoding.cmake:12]: If the mentioned test case means https://github.com/doxygen/doxygen/files/11578931/example.tar.gz, this PR local build doxygen generate html with my expected output:

图片

Comment (albert-github) [reply cmake/WindowsEncoding.cmake:12]: And for Bulgarian and English?

Comment (zchrissirhcz) [reply cmake/WindowsEncoding.cmake:12]: Bulgarian and English are also correct. Re-compile takes some time (BTW, speed up compilation of doxygen may be considered.)

## Confirm version
Re-compiled on my machine, confirm versions:

### git commit: 
* a0441205e 2023-05-27 | Set charset for source and execution on Windows (HEAD -> windows_encoding, origin/windows_encoding) [Zhuo Zhang]

### doxygen version:
C:\Users\zz\github\doxygen\build\bin>.\doxygen --version
1.9.8 (a0441205e6337f58278f985769a976b71a587afe)

### git status
```
C:\Users\zz\github\doxygen\build\bin>git status
On branch windows_encoding
Your branch is up to date with 'origin/windows_encoding'.

nothing to commit, working tree clean
```

## Screenshot
Bulgarian:
bulgarian

English:
english

**Comment by zchrissirhcz** [comment]
https://github.com/doxygen/doxygen/actions/runs/5098341489/jobs/9165423331?pr=10095
The Windows Latest MSVC Release CI pipeline failed, GraphViz failed to install by Chocolatey time out. Maybe an cache step can be introduced to the github action workflows to save time.

File: cmake/WindowsEncoding.cmake (hunk 1 above)
Comment (albert-github) [reply cmake/WindowsEncoding.cmake:12]: Thanks these images look good.

**Comment by albert-github** [comment]
I just restarted the build (as there were already 6 completed I don't expect any problems with the 2 restarted ones)

File: cmake/QueryCodePage.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import platform
+
+def is_windows():
+    return platform.system().lower() == "windows"
+
+if is_windows():
+    from winreg import (
+        CloseKey, OpenKey, QueryValueEx, SetValueEx,
```
Comment (doxygen) [inline cmake/QueryCodePage.py:8]: I guess the list of imports can be trimmed a bit. Doesn't look like `CloseKey`, `SetValueEx`, `HKEY_CURRENT_USER`, `KEY_ALL_ACCESS`, `REG_EXPAND_SZ`, `REG_SZ` are used.

Comment (zchrissirhcz) [reply cmake/QueryCodePage.py]: Yes they are not used imports. Now I correct them as suggested, force pushed.

**Pull request merged**